### ISSUE TITLE
Lots of crashes on `FlushRequest.sendRequest(_:type:useIP:completion:)` with 4.0.4

### ISSUE THREAD

**Issue by malcommac** [mixpanel/mixpanel-swift#578]
**Integration Method:** SPM
**Xcode Version:** 14.1.0
**Library Version:** 4.0.4
**Platform:** iOS
**Language:** Swift
**Description:** 

We got lots of crashes using the 4.0.4 release on `FlushRequest.sendRequest(_:type:useIP:completion:)`:

<img width="739" alt="image" src="https://user-images.githubusercontent.com/235645/208411373-150746f0-0c1a-4364-b921-ca2f9b7d3110.png">

The raw stack trace:

- From Crashlytics: [1000_issue_bdea1234c542cc5311cd883d0eaa07fe_crash_session_7f2766abffd840518ee45bbfe240ac1f_DNE_0_v2_stacktrace.txt](https://github.com/mixpanel/mixpanel-iphone/files/10258669/1000_issue_bdea1234c542cc5311cd883d0eaa07fe_crash_session_7f2766abffd840518ee45bbfe240ac1f_DNE_0_v2_stacktrace.txt)
- From Sentry: [938f83131e154ff9b6d9fcbacb151413-symbolicated.txt](https://github.com/mixpanel/mixpanel-swift/files/10400468/938f83131e154ff9b6d9fcbacb151413-symbolicated.txt)

**Comment by malcommac** [comment]
anyone working on this SDK anymore?

**Comment by jaredmixpanel** [comment]
@malcommac Are you able to reproduce this issue? If so, can you provide the steps? Were you using older versions without this problem? If so, what version were you upgrading from? Any additional info you can provide would be helpful.

**Comment by malcommac** [comment]
Before that, we used 3.5.1 without the issue. Unfortunately, I'm not able to reproduce this issue, but it seems to be happening at the early stage of the app launch. Maybe it could be related to some flush events of the local events not sent in the previous session?

**Comment by malcommac** [comment]
I wonder if anyone read these issues. Anyway it still the most relevant crash issue in our app.


[1000_issue_bdea1234c542cc5311cd883d0eaa07fe_crash_session_81cd6794d25a40fda67827a318c44298_DNE_0_v2_stacktrace.txt](https://github.com/mixpanel/mixpanel-swift/files/11231687/1000_issue_bdea1234c542cc5311cd883d0eaa07fe_crash_session_81cd6794d25a40fda67827a318c44298_DNE_0_v2_stacktrace.txt)

**Comment by jaredmixpanel** [comment]
@malcommac Apologies for the lack of progress here but I was never able to reproduce this crash and this is the only report we have. If you get additional information, particularly anything that might help reproduce the issue please add it here.

**Comment by pmgsPT** [comment]
Hello,
We are also having this crash since we integrated Mixpanel. It is currently the top crash in our app. We also noticed it happens at the first second of opening the app.

**Comment by yeraydavidrodriguez** [comment]
Hello @jaredmixpanel. Have Mixpanel plans for checking this? Because it is the first cause of crashes in our app, and as you see we are not the only ones affected. Thanks!

**Comment by jaredmixpanel** [comment]
@malcommac @yeraydavidrodriguez @pmgsPT @gfoldv Just released `v4.1.4` with a possible fix for this issue. Once you've updated please report back and let us know the results.

**Comment by malcommac** [comment]
I'll try and let u know asap, thx

**Comment by gfoldv** [comment]
Still an issue in the v4.1.4
Pretty much the same stacktrace

```
Crashed: com.mixpanel.98b4601e9379667d3d2e678444373306.network)
0  <redacted>                    0xa75810 FlushRequest.sendRequest(_:type:useIP:) + 30 (Network.swift:30)
1  <redacted>                    0xa74db8 Flush.flushQueueInBatches(_:type:) + 107 (Flush.swift:107)
2  <redacted>                    0xa94ae8 closure #1 in closure #1 in MixpanelInstance.flush(performFullFlush:completion:) + 973 (MixpanelInstance.swift:973)
3  <redacted>                    0xa8c428 thunk for @escaping @callee_guaranteed () -> () + 4388946984 (<compiler-generated>:4388946984)
4  libdispatch.dylib              0x26a8 _dispatch_call_block_and_release + 32
5  libdispatch.dylib              0x4300 _dispatch_client_callout + 20
6  libdispatch.dylib              0xb894 _dispatch_lane_serial_drain + 748
7  libdispatch.dylib              0xc3f8 _dispatch_lane_invoke + 432
8  libdispatch.dylib              0x17004 _dispatch_root_queue_drain_deferred_wlh + 288
9  libdispatch.dylib              0x16878 _dispatch_workloop_worker_thread + 404
10 libsystem_pthread.dylib        0x1964 _pthread_wqthread + 288
11 libsystem_pthread.dylib        0x1a04 start_wqthread + 8
```

Please note, that the crash happens in the Network.swift -> line 30

**Comment by sipersso** [comment]
Same issue in 4.1.4

**Comment by pmgsPT** [comment]
Hello, any update here?
I updated the library and it the crash is still there...still the top crash in our app.

**Comment by pavel-trafimuk** [comment]
Hi, we also updated to latest mixpanel build at your company and this crash appears in top-5 crashes even in small roll-out. We stopped release with trying to understand how to fix it

**Comment by 3a4oT** [comment]
Hey, same here. This is a top crash for us.

**Comment by AminHeidariD** [comment]
Same issue on version 4.1.4 and also top crash for us

**Comment by slipdef** [comment]
Is it going to be fixed ever? We have 3-5 crashes everyday with it.

**Comment by gfoldv** [comment]
@jaredmixpanel @zihejia how can we assist you to find the cause of the crash?

**Comment by zihejia** [comment]
hi @malcommac @gfoldv , the root cause of the crash was a race condition occurring when setting the server URL. We have addressed this issue in [v4.2.6](https://github.com/mixpanel/mixpanel-swift/releases/tag/v4.2.6). I am closing this issue, but please feel free to reopen it if the problem persists.

**Comment by gfoldv** [comment]
@zihejia @jaredmixpanel this is still an issue in 4.2.7
same error as before.
@malcommac are you experiencing the crash?
